### PR TITLE
More error reporting from the operation store

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/op_store.rs [hunk 1]
```diff
@@ -381,7 +381,7 @@ pub enum OpStoreError {
     #[error("Operation not found")]
     NotFound,
     #[error("{0}")]
-    Other(String),
+    Other(Box<dyn std::error::Error + Send + Sync>),
```
Comment (yuja) [inline lib/src/op_store.rs:332]: nit: can be `#[error(transparent)]`

File: lib/src/backend.rs [hunk 2]
```diff
@@ -43,7 +43,7 @@ macro_rules! id_type {
             #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Hash)]
             $vis struct $name(Vec<u8>);
         }
-        impl_id_type!($name);
+        crate::backend::impl_id_type!($name);
```
Comment (yuja) [inline lib/src/backend.rs:46]: nit: I slightly prefer `$crate` (though it wouldn't matter for crate-local macros)

And let's fully spell `std::fmt::Debug`/`Formatter`/`Error` to not rely on names imported at the call site.

**Review by yuja — APPROVED** [review]
Nice!

File: lib/src/backend.rs (hunk 2 above)
Comment (martinvonz) [reply lib/src/backend.rs]: Good idea. Done.

**Pull request merged**